lib/awful/layout/init.lua.in

```diff
@@ -109,7 +109,15 @@ function layout.arrange(screen)
     p.geometry = capi.screen[screen].geometry
     p.clients = client.tiled(screen)
     p.screen = screen
-    layout.get(screen).arrange(p)
+
+    -- Apply layout geometry
+    local gs = layout.get(screen).arrange(p)
```

Comment:
The problem is that we sometime terminal the object before is is GCed, so you get "invalid user data" error. It's an awesome bug. Then, the client can be killed while arrange is called, imagine a tag with 1000 short lived clients. Who know